core/src/main/scala/org/sgine/component/prop/PreferredSize.scala

```diff
@@ -3,9 +3,12 @@ package org.sgine.component.prop
 import pl.metastack.metarx.{ReadStateChannel, Var}
 
 class PreferredSize {
-  private[sgine] val _width = Var[Double](0.0)
-  private[sgine] val _height = Var[Double](0.0)
+  private val _width = Var[Double](0.0)
+  private val _height = Var[Double](0.0)
 
   def width: ReadStateChannel[Double] = _width
   def height: ReadStateChannel[Double] = _height
+
+  def width(value: Double): Unit = _width := value
+  def height(value: Double): Unit = _height := value
```

Comment:
The reason I didn't do this is because I only wanted Sgine to be able to modify the width and height.  With your updates there's nothing stopping someone using the framework from calling this and changing the width and height.